### PR TITLE
[CIS-1354] Pipeline to distribute Static Libraries

### PR DESCRIPTION
**Pull request by polqf**
### 🔗 Issue Link
https://stream-io.atlassian.net/browse/CIS-1354

### 🎯 Goal

Be able to distribute our code as static libraries

### 🛠 Implementation

This PR adds a script that will create static libraries wrapped in xcframeworks.
It also creates 2 bundle files that contain the needed resources for each library.

Detailed explanation here: https://www.notion.so/stream-chat-fe-sdk/Static-Libraries-4c5c4220b1fa45a1bbd7d52f5ce3f585

Screenshot 2021-11-24 at 13 20 32

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

File: Sources/StreamChat/Utils/Bundle+Extensions.swift [hunk 1]
```diff
@@ -39,6 +39,10 @@ extension Bundle {
             .flatMap(Bundle.init(url:))!
         #elseif SWIFT_PACKAGE
         return Bundle.module
+        #elseif STATIC_LIBRARY
```
Comment (polqf) [inline Sources/StreamChat/Utils/Bundle+Extensions.swift:42]: Explanation here: https://www.notion.so/stream-chat-fe-sdk/Static-Libraries-4c5c4220b1fa45a1bbd7d52f5ce3f585#bd3e2b1fe552471a9bc99d31cea92474

File: Makefile [hunk 2]
```diff
@@ -3,24 +3,65 @@ MAKEFLAGS += --silent
...
-	bundle exec fastlane build_xcframeworks
-	echo "♦︎♦︎♦︎ Creating compressed archives ♦︎♦︎♦︎"
-	cd ./Products && zip -r "StreamChat-All.xcframework.zip" ./*.xcframework
-	cd ./Products && zip -r "StreamChat.xcframework.zip" ./StreamChat.xcframework
-	cd ./Products && zip -r "StreamChatUI.xcframework.zip" ./StreamChatUI.xcframework
-	echo "♦︎♦︎♦︎ Checksum for StreamChat.xcframework.zip ♦︎♦︎♦︎"
-	swift package compute-checksum Products/StreamChat.xcframework.zip
-	echo "♦︎♦︎♦︎ Checksum for StreamChatUI.xcframework.zip ♦︎♦︎♦︎"
-	swift package compute-checksum Products/StreamChatUI.xcframework.zip
+all_artifacts:
```
Comment (adamrushy) [inline Makefile:6]: This script is 🤩 😍

**Review by adamrushy — APPROVED** [review]
All looks good to me 👍🏼

**Review by nuno-vieira — APPROVED** [review]
LGTM! 🚀 Really good job 👌

**Review by b-onc — APPROVED** [review]

File: Scripts/buildStaticLibraries.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+          -derivedDataPath $derived_data \
+          -archivePath "$archives/$library-iphone-$platform" \
+          SKIP_INSTALL=NO \
+          BUILD_LIBRARY_FOR_DISTRIBUTION=YES > $logs_file
+
+  # Unlike `xcodebuild build`, `xcodebuild archive` does not generate a .swiftmodule for static libraries. So we manually
+  # copy it from our local DerivedData
+  #
+  # Expanded explanation:
+  # We cannot use `xcodebuild build` artifacts because those are build with the flag -fembed-bitcode-marker.
```
Comment (bielikb) [inline Scripts/buildStaticLibraries.sh:57]: Hey @polqf,
wouldn't adding the appropriate env var overrides to `xcodebuild build` resolve the issue around the bitcode?

eg
```
xcodebuild build \
....
....
OTHER_CFLAGS="-fembed-bitcode" \ 
BITCODE_GENERATION_MODE="bitcode" \ 
ENABLE_BITCODE=YES
```

?

Comment (polqf) [reply Scripts/buildStaticLibraries.sh:57]: I tried, but passing -fembed-bitcode in OTHER_CFLAGS does not take precedence over the already defined -fembed-bitcode-marker in xcodebuild 🤷‍♂️ .

**Review by bielikb — APPROVED** [review]

**Pull request merged**